src/Dropdown/Dropdown.test.tsx

```diff
@@ -1,213 +1,820 @@
+        const props: DropdownProps = {
+            ...DEFAULT_PROPS,
+            multi: true,
+            selectedValue: [{ ...DEFAULT_PROPS.list[0] }]
+        };
+
+        act(() => {
+            render(<Dropdown {...props} />, container);
+        });
+
+        const closeButton: Element = container.querySelector(".dropdown-times-icon");
+        expect(closeButton).toBeTruthy();
+
+        const newProps: DropdownProps = {
+            ...DEFAULT_PROPS,
+            clearable: true,
+            selectedValue: { ...DEFAULT_PROPS.list[0] }
+        };
+
+        act(() => {
+            render(<Dropdown {...newProps} />, container);
+        });
+
+        expect(closeButton).toBeTruthy();
     });
 
     it("Should not display clear button if all items are unselected even when clearable prop set to true", () => {
-        const wrapper = mount(
-            <Dropdown
-                {...props}
-                clearable={true}
-            />
-        );
-        expect(wrapper.find(".dropdown-times-icon").length).toBe(0);
-        wrapper.unmount();
+        const props: DropdownProps = {
+            ...DEFAULT_PROPS,
+            clearable: true
+        };
+
+        act(() => {
+            render(<Dropdown {...props} />, container);
+        });
+
+        const closeButton: Element = container.querySelector(".dropdown-times-icon");
+        expect(closeButton).toBeFalsy();
+    });
+
+    it("Should clear all items selected when clear button pressed", () => {
+        const props: DropdownProps = {
+            ...DEFAULT_PROPS,
+            selectedValue: { ...DEFAULT_PROPS.list[0] },
+            clearable: true
+        };
+
+        act(() => {
+            render(<Dropdown {...props} />, container);
+        });
+
+        const target: Element = container.querySelector(".right-items > .dropdown-icon-holder");
+        expect(target).toBeTruthy();
+
+        act(() => {
+            target.dispatchEvent(new MouseEvent("click", { bubbles: true }));
+        });
+
+        expect(props.onChange).toBeCalledTimes(1);
+        expect(props.onChange).toBeCalledWith(null);
+
+        act(() => {
+            render(<Dropdown {...props} selectedValue={null} />, container);
+        });
+
+        const target2: Element = container.querySelector(".right-items > .dropdown-icon-holder > .dropdown-times-icon");
+        expect(target2).toBeFalsy();
+    });
+
+    it("Should select all items when select all button pressed", () => {
+        const props: DropdownProps = {
+            ...DEFAULT_PROPS,
+            selectedValue: [{ ...DEFAULT_PROPS.list[0] }],
+            multi: true
+        };
+
+        act(() => {
+            render(<Dropdown {...props} />, container);
+        });
+
+        const target: Element = container.querySelector("#select-all");
+        expect(target).toBeTruthy();
+
+        const selectedOptions: NodeListOf<Element> = container.querySelectorAll(".custom-dropdown-item.selected");
```

Comment:
You aren't reusing anywhere , why the variable ?

Reply:
you're right